src/category_theory/simplex_category.lean

```diff
@@ -0,0 +1,230 @@
+  { simp at h_1, linarith, },
+  { exfalso, exact lt_irrefl _ (lt_of_le_of_lt (nat.le_pred_of_lt (nat.lt_of_succ_le h)) h_3), },
+  { exact (nat.succ_pred_eq_of_pos (lt_of_le_of_lt (zero_le _) h_2)).symm, },
+end
+
+local attribute [simp] fin.pred_mk
+
+/-- The fifth simplicial identity -/
+lemma σ_comp_σ {n} {i j : fin (n+1)} (H : i ≤ j) :
+  σ i.cast_succ ≫ σ j = σ j.succ ≫ σ i :=
+begin
+  ext k,
+  dsimp [σ, fin.pred_above],
+  rcases i with ⟨i, _⟩,
+  rcases j with ⟨j, _⟩,
+  rcases k with ⟨k, _⟩,
+  simp at H,
+  simp with push_cast,
+  split_ifs,
+  -- Hope for the best from `linarith`:
+  any_goals { simp at *, try { linarith }, },
+  { exact false.elim
+    (lt_irrefl (k - 1)
+      (lt_of_lt_of_le (nat.pred_lt (id (ne_of_lt (lt_of_le_of_lt (zero_le i) h)).symm))
+        (le_trans h_2 (nat.succ_le_of_lt h_1)))) },
+  { exact false.elim
+    (lt_irrefl j (lt_of_lt_of_le (nat.pred_lt_pred (nat.succ_ne_zero j) h_2) h_1)) },
+end
+
+end generators
+
+section skeleton
+
+/-- The functor that exhibits `simplex_category` as skeleton
+of `NonemptyFinLinOrd` -/
+def skeletal_functor :
+  simplex_category ⥤ NonemptyFinLinOrd.{u} :=
+{ obj := λ n, NonemptyFinLinOrd.of $ ulift (fin (n+1)),
+  map := λ m n f, ⟨λ i, ⟨f i.down⟩, λ ⟨i⟩ ⟨j⟩ h, show f i ≤ f j, from f.monotone h⟩, }
+
+namespace skeletal_functor
+
+instance full : full.{0 u 0 (u+1)} skeletal_functor.{u} :=
+{ preimage := λ m n f, ⟨λ i, (f ⟨i⟩).down, λ i j h, f.monotone h⟩,
+  witness' := by { intros m n f, dsimp at *, ext1 ⟨i⟩, ext1, refl } }
+
+instance : faithful skeletal_functor.{u} :=
+{ map_injective' := λ m n f g h,
+  begin
+    ext1 i, apply equiv.ulift.symm.injective,
+    show skeletal_functor.map f ⟨i⟩ = skeletal_functor.map g ⟨i⟩,
+    rw h,
+  end }
+
+instance : ess_surj skeletal_functor.{u} :=
+{ mem_ess_image := λ X, ⟨(fintype.card X - 1 : ℕ), ⟨begin
+    have aux : fintype.card X = fintype.card X - 1 + 1,
+    { exact (nat.succ_pred_eq_of_pos $ fintype.card_pos_iff.mpr ⟨⊥⟩).symm, },
+    let f := mono_equiv_of_fin X aux,
+    have hf := (finset.univ.order_emb_of_fin aux).strict_mono,
+    refine
+    { hom := ⟨λ i, f i.down, _⟩,
+      inv := ⟨λ i, ⟨f.symm i⟩, _⟩,
+      hom_inv_id' := _,
+      inv_hom_id' := _ },
+    { rintro ⟨i⟩ ⟨j⟩ h, show f i ≤ f j, exact hf.monotone h, },
+    { intros i j h, show f.symm i ≤ f.symm j, rw ← hf.le_iff_le,
+      show f (f.symm i) ≤ f (f.symm j), simpa only [order_iso.apply_symm_apply], },
+    { ext1 ⟨i⟩, ext1, exact f.symm_apply_apply i },
+    { ext1 i, exact f.apply_symm_apply i },
+  end⟩⟩,}
+
+noncomputable instance is_equivalence : is_equivalence skeletal_functor.{u} :=
+equivalence.equivalence_of_fully_faithfully_ess_surj skeletal_functor
+
+end skeletal_functor
+
+/-- The equivalence that exhibits `simplex_category` as skeleton
+of `NonemptyFinLinOrd` -/
+noncomputable def skeletal_equivalence :
+  simplex_category ≌ NonemptyFinLinOrd.{u} :=
+functor.as_equivalence skeletal_functor.{u}
```

Comment:
You can add `is_skeleton_of` here potentially

Reply:
Ah! Good suggestion. Done.